### ISSUE TITLE
InRelease sig verification fails on Debian 11, halting 'apt update'

### ISSUE THREAD

**Issue by deutrino** [syncthing/syncthing#8550]
This issue is actually preventing me from installing Syncthing, and I verified it on a host that already had latest stable installed. The issue is now preventing ``apt update`` on that machine as well unless I remove the Syncthing repo from apt sources.

This wasn't happening 2 days ago when I was working with these machines so something has changed. But for now it looks (from my vantage point) like install & update of Syncthing from ``apt.syncthing.net`` might be broken for all Debian 11 machines??

```
# apt update
Hit:1 http://deb.debian.org/debian bullseye InRelease
Get:2 http://deb.debian.org/debian bullseye-updates InRelease [44.1 kB]                   
Get:3 http://security.debian.org/debian-security bullseye-security InRelease [48.4 kB]    
Get:4 http://apt.syncthing.net syncthing InRelease [15.1 kB]                              
[snip]
Err:4 http://apt.syncthing.net syncthing InRelease             
  The following signatures were invalid: BADSIG D26E6ED000654A3E Syncthing Release Management <release@syncthing.net>
Get:9 http://deb.debian.org/debian bullseye-backports/main amd64 Packages.diff/Index [63.3 kB]
[snip]
W: An error occurred during the signature verification. The repository is not updated and the previous index files will be used. GPG error: http://apt.syncthing.net syncthing InRelease: The following signatures were invalid: BADSIG D26E6ED000654A3E Syncthing Release Management <release@syncthing.net>
W: Failed to fetch http://apt.syncthing.net/dists/syncthing/InRelease  The following signatures were invalid: BADSIG D26E6ED000654A3E Syncthing Release Management <release@syncthing.net>
W: Some index files failed to download. They have been ignored, or old ones used instead.

```

**Comment by deutrino** [comment]
Just to add I am running a caching proxy server in front of apt, but tried both ``http`` and ``https`` in the apt sources, with the same signature problem. So I doubt my ``apt-cacher-ng`` instance is the issue. But I will try disabling it and see if anything changes.

**Comment by calmh** [comment]
Yeah, the signature is fine as far as I can see, so I think you'll need to debug this locally.

```
jb@sep:~ % curl -sO https://apt.syncthing.net/dists/syncthing/InRelease
jb@sep:~ % gpg --verify InRelease
gpg: Signature made Wed Sep 14 11:06:00 2022 CEST
gpg:                using RSA key D26E6ED000654A3E
gpg: Good signature from "Syncthing Release Management <release@syncthing.net>" [ultimate]
```

**Comment by deutrino** [comment]
Interestingly, when I came back to this today, I was able to reproduce the problem on one of the affected machines but not the other.

I did a little more digging on this and found an ``InRelease`` file in ``apt-cacher-ng``'s cache that did indeed have a bad signature. It's not visually obvious to me that there's anything wrong with the file (no weird whitespace or obvious corruption).

Anyway I preserved a copy of that file in case it's needed, then flushed it from the cache - I'm using ``apt-cacher-ng`` version 3.7.4-1~bpo11+1 - and after doing this, was no longer able to reproduce the problem, and successfully updated to latest Syncthing on a machine where I'd previously had to use the version in bullseye-backports due to this problem.

No idea what may have happened, perhaps a glitch or bug in ``apt-cacher-ng`` - although I did a ``diff`` of the current ``InRelease`` file (downloaded with the same ``curl`` command as in the previous comment) and the bad one, and noted that both the signature, and many of the checksums, differed between the two files.
